src/genworld.cpp

```diff
@@ -337,3 +341,157 @@ void GenerateWorld(GenWorldMode mode, uint size_x, uint size_y, bool reset_setti
 
 	_GenerateWorld();
 }
+
+/** Town data imported from JSON files and used to place towns. */
+struct ExternalTownData {
+	TownID town_id; ///< The TownID of the town in OpenTTD. Not imported, but set during the founding proceess and stored here for convenience.
+	std::string name; ///< The name of the town.
+	uint population; ///< The target population of the town when created in OpenTTD. If input is blank, defaults to 0.
+	bool is_city; ///< Should it be created as a city in OpenTTD? If input is blank, defaults to false.
+	float x_proportion; ///< The X coordinate of the town, as a proportion 0..1 of the maximum X coordinate.
+	float y_proportion; ///< The Y coordinate of the town, as a proportion 0..1 of the maximum Y coordinate.
+};
+
+/**
+ * Helper for CircularTileSearch to found a town on or near a given tile.
+ * @param tile The tile to try founding the town upon.
+ * @param user_data The ExternalTownData to attempt to found.
+ * @return True if the town was founded successfully.
+ */
+static bool TryFoundTownNearby(TileIndex tile, void *user_data)
+{
+	ExternalTownData &town = *static_cast<ExternalTownData *>(user_data);
+	std::tuple<CommandCost, Money, TownID> result = Command<CMD_FOUND_TOWN>::Do(DC_EXEC, tile, TSZ_SMALL, town.is_city, _settings_game.economy.town_layout, false, 0, town.name);
+
+	TownID id = std::get<TownID>(result);
+
+	/* Check if the command failed. */
+	if (id == INVALID_TOWN) return false;
+
+	/* The command succeeded, send the ID back through user_data. */
+	town.town_id = id;
+	return true;
+}
+
+/**
+ * Load town data from _file_to_saveload and place its contents into the world.
+ * @return True if the JSON file is valid and has north and south extents, otherwise false.
```

Comment:
Nitpick: if you write `True if ..., otherwise false`, just write: `True iff ...`. Saves the reader some time not having to read the `otherwise false` non-sense :P :P